libs/checkpoint/src/store/memory.ts

```diff
@@ -113,35 +222,224 @@ export class InMemoryStore extends BaseStore {
 
     namespaces.sort((a, b) => a.join(":").localeCompare(b.join(":")));
 
-    const paginatedNamespaces = namespaces.slice(
-      op.offset,
-      op.offset + op.limit
+    return namespaces.slice(
+      op.offset || 0,
+      (op.offset || 0) + (op.limit || namespaces.length)
     );
-
-    return paginatedNamespaces;
   }
 
   private doesMatch(matchCondition: MatchCondition, key: string[]): boolean {
     const { matchType, path } = matchCondition;
 
-    if (key.length < path.length) {
-      return false;
-    }
-
     if (matchType === "prefix") {
+      if (path.length > key.length) return false;
       return path.every((pElem, index) => {
         const kElem = key[index];
         return pElem === "*" || kElem === pElem;
       });
     } else if (matchType === "suffix") {
-      return [...path].reverse().every((pElem, index) => {
-        const kElem = key[key.length - 1 - index];
+      if (path.length > key.length) return false;
+      return path.every((pElem, index) => {
+        const kElem = key[key.length - path.length + index];
         return pElem === "*" || kElem === pElem;
       });
     }
 
     throw new Error(`Unsupported match type: ${matchType}`);
   }
+
+  private filterItems(op: SearchOperation): Item[] {
```

Comment:
Should we instead make the base class take a `VectorStore` instance?

We could easily port this:

https://js.langchain.com/docs/integrations/vectorstores/memory/

Reply:
We could maybe extend support, but right now the data models don't really align